code/backend/batch/utilities/helpers/processors/DocumentProcessorHelper.py

```diff
@@ -0,0 +1,56 @@
+import logging
+from typing import List
+
+from ..AzureBlobStorageClient import AzureBlobStorageClient
+
+from ..Processor import Processor
+from ..ConfigHelper import ConfigHelper
+
+from .ProcessorBase import ProcessorBase
+from ..AzureSearchHelper import AzureSearchHelper
+from ..DocumentLoadingHelper import DocumentLoading
+from ..DocumentChunkingHelper import DocumentChunking
+from ...common.SourceDocument import SourceDocument
+
+logger = logging.getLogger(__name__)
+
+
+class DocumentProcessor(ProcessorBase):
+    def __init__(self, blob_client: AzureBlobStorageClient):
+        self.blob_client = blob_client
+
+    def process_file(self, source_url: str, file_name: str):
+        file_extension = file_name.split(".")[-1]
+        processors = [
+            x
+            for x in ConfigHelper.get_active_config_or_default().document_processors
+            if x.document_type.lower() == file_extension.lower()
+        ]
```

Comment:
thinking aloud here::
1. it seems whenever we have to process a file, we call this fxn
2. we load the configs from the helper or get default
3. iterate over them each time and get a matching one
4. then process.

Can we rather:
1. on init  of this class, do the step(3) above once and store in a dictionary/map/loop-up
2. So, whenever this fxn is called, we dont do iteration each time, but,just do a constant O(1) lookup in the dictionary based on the extension ?

Reply:
👍 Updated